sw/host/opentitanlib/src/test_utils/mem.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use std::time::Duration;
+
+use crate::io::uart::Uart;
+use crate::test_utils::e2e_command::TestCommand;
+use crate::test_utils::rpc::{UartRecv, UartSend};
+use crate::test_utils::status::Status;
+
+// Bring in the auto-generated sources.
+include!(env!("mem"));
+
+impl MemRead32Req {
+    pub fn execute(&self, uart: &dyn Uart) -> Result<MemRead32Resp> {
+        TestCommand::MemRead32.send(uart)?;
+        self.send(uart)?;
+        let resp = MemRead32Resp::recv(uart, Duration::from_secs(300), false)?;
+        Ok(resp)
+    }
+}
+
+impl MemReadReq {
```

Comment:
For those as well, having a constructor might make sense. Also probably a higher-level  interface that splits a transfer in chunks of 256 would be helpful because it's really not obvious with the rust structure being generated from the C code. You could consider changing `execute` to
```rust
pub fn execute(uart: &dyn Uart, addr: u32, size: usize) -> Result<Vec<u8>> {
   // do a loop, blablabla
}

Reply:
You want me to make usable APIs instead of the bare minimum? Psh... fine... ;)

Done, but I went with more of a slice pattern.